### PR TITLE
Guncargo 2023 : I love the free market : No interest, no buying companies, no money down! : Cheaper literally everything

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

Something more descriptive will be put here eventually but basically what the title says:

No company interest requirements
No buying companies to access their goods
Everything is a hell of a lot cheaper
Borgs can use the imports menu
Item stock is gone you just buy stuff
Also orders come in single cases or crates again (unless cargo orders them because they cant be goodies anymore)

Soon to come:

I change the items lists all over again

Now, while items ordered normally through cargo still show up in singular boxes, these boxes are special wooden boxes with no export value to make sure you cant make infinite cash off of ordering sutures.

## How This Contributes To The Skyrat Roleplay Experience

Have you ever wanted to buy something but god said you must grind a gorillon bounties to be able to get interest high enough? Yeah not anymore.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image
image

Note that I removed a bunch of stuff before the research director screenie to see if that worked (it did)
image
image

</details>

<details>
<summary>Real caught footage of a borg ordering stuff through imports</summary>

image
image

</details>

## Changelog

:cl:
qol: When ordering multiple items with a personal account or with a department budget, all of those items will arrive in the same container, case or crate depending on the number of items.
balance: Company imports companies no longer need to be purchased
balance: Company interest no longer exists at all
balance: Company import items cannot run out of stock anymore
balance: Company import items are overall significantly cheaper than they used to be
balance: Some items in company imports have either been added, removed, or tweaked in nearly all companies due to not quite fitting anymore, or missing where I felt they should have been. Most of the removed items are comically easy to attain or otherwise useless and were only there to raise interest before, but now that interest is gone they are not necessary.
fix: Cyborgs can use company imports to order things off of the cargo budget now
image: Added a new wooden box sprite for import items that come from the cargo budget, to prevent infinite money generation at the same time
/:cl:

### CODE REVIEW

**Comment by SpaceLoveSs13** [comment]
This is not bad, the reasons brought forward are reasonable and I personally have experienced the pain of grinding money away just to see interest go up. And that imo serves no point besides wasting hard earned credits and time. ~~The rate of stock refilling should also be increased to compensate the change though!~~ I re-read the code and noticed that part of it is entirely gone. Even better!

**Comment by burgerenergy** [comment]
I mentioned it in discord but I'll mention it here, too; I am kind of concerned this might lead to weirdness with command budgets - when they actually fucking use it - buying up loads of stuff at the start given there is no scarcity and nothing to use that budget on BUT buying stuff at cargo. Particularly, security and engineering can bypass science to get MCR/tool doodads immediately, all for no opportunity cost. Is this bad? Eh.

If this is a concern I think a fix could be _having_ stocks that regenerate overtime, or just living with it. It's more convenient to have it from the lathe, so maybe it'll be fine.

**Comment by Paxilmaniac** [comment]
> I mentioned it in discord but I'll mention it here, too; I am kind of concerned this might lead to weirdness with command budgets - when they actually fucking use it - buying up loads of stuff at the start given there is no scarcity and nothing to use that budget on BUT buying stuff at cargo. Particularly, security and engineering can bypass science to get MCR/tool doodads immediately, all for no opportunity cost. Is this bad? Eh.
> 
> If this is a concern I think a fix could be _having_ stocks that regenerate overtime, or just living with it. It's more convenient to have it from the lathe, so maybe it'll be fine.

The intention is that heads can choose to spend their budget and thus their underling’s future paychecks on having certain advanced items early. Alternatively, if someone doesn’t want to wait for science or their head to get some of the advanced stuff, they can spend some of their own money.

The list of items is going to change some since a lot of the entries in certain categories is only there for when interest was still around, but this is certainly built around not only individuals buying stuff but also heads bulk ordering things to make the lives of their staff a bit easier.

**Comment by Paxilmaniac** [comment]
oops forgot a changelog

**Comment by ghost** [comment]
Perhaps if some items DO present problems in the future, they can be locked behind time gates or something similar? Preventing people skipping Science altogether but not removing the benefit of going to Cargo to get around absent Scientists?
Otherwise, I'm all for this, besides one quick clarification:
Will these imports be subjected to department access locks? Making guns available roundstart for anybody with the credits seems like it'll ruin the entire time scaling meant to tie into Dynamic and the likes, and will make any early midrounds or lighter OpFors severely outgunned...

Aside from those two concerns though, I'd legitimately say you should see if upstream would accept this. One of the drawbacks of cargo is how you can only get bulk packs of stuff that you don't even want all of, I.E. medical, so being able to just purchase a new stock of meshes/sutures specifically would be a huge boon

**Comment by tf-4** [comment]
But Why.
"I want thing without timelock" isn't exactly justification.

**Comment by Paxilmaniac** [comment]
https://youtu.be/xyVXl2ALgec

**Comment by chemistrymain2** [comment]
> But Why. "I want thing without timelock" isn't exactly justification.

We don't need timelocks, we're not TGstation.

Nobody likes cargo imports to begin with-nobody likes the import system, nobody likes 'company interest', so on, so forth.

It's a thing, it's in cargo, it should, ergo, be available in cargo.

_Roundstart balance is not a concern._

**Comment by burgerenergy** [comment]
Company interest really is just tedium and forcing cargo to buy random things for the sake of getting more random things. That going away is a net positive. It has the side issue of some weapons being available from the get-go, but eh. If it's an issue the price can be adjusted.

**Comment by Plum-but-gayer** [comment]
ITEM STOCK BEING GONE IS HUGE
now atta can finally bulk buy allstar SC-1s to make tower defense turrets during spiderhell rounds. Thank you!!
I think timelocks are fine, I think company interest is just a bad system bc waiting for someone to take the 'fall' of buying some cheap useless stuff first just encourages more grinding. In general bounties are pretty rote and don't add to RP, so the less time people spend 'earning' toys and the more time we have playing with them is good.

File: modular_skyrat/modules/company_imports/code/datums/cargo_pack.dm [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #define CARGO_CUT 0.05
 
 /datum/supply_pack/armament
+	goody = TRUE
+	crate_type = /obj/structure/closet/crate/large/import
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/company_imports/code/datums/cargo_pack.dm:5]: Won't this cause crates that are worth money to be sent through goodies?

File: tgui/packages/tgui/interfaces/CargoImportConsole.js [hunk 2]
```diff
@@ -147,49 +117,26 @@ export const CargoImportConsole = (props, context) => {
...
-                            {!!item.buyable_ammo && (
-                              <Stack.Item
-                                textColor={
-                                  item.magazine_cost > budget_points
-                                    ? 'red'
-                                    : 'green'
-                                }>
-                                {'Ammo Cost: ' + item.magazine_cost}
-                              </Stack.Item>
-                            )}
```
Comment (GoldenAlpharex) [inline tgui/packages/tgui/interfaces/CargoImportConsole.js:183]: So you can't buy the ammo anymore?

File: modular_skyrat/modules/company_imports/code/armament_datums/kahraman_industries.dm [hunk 3]
```diff
@@ -5,47 +5,31 @@
 // Mining PPE, SEVAs and hardhats, have you passed your OSHA inspection today?
 
 /datum/armament_entry/company_import/kahraman/ppe
-	subcategory = "OSHA Certified Protective Equipment"
-
-/datum/armament_entry/company_import/kahraman/ppe/hardhat
-	item_type = /obj/item/clothing/head/utility/hardhat/orange
-	lower_cost = CARGO_CRATE_VALUE * 0.25
-	upper_cost = CARGO_CRATE_VALUE
+	subcategory = "FOHSA Certified Protective Equipment"
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/company_imports/code/armament_datums/kahraman_industries.dm:8]: Why the change?

File: modular_skyrat/modules/company_imports/code/armament_datums/deforest_medical.dm [hunk 4]
```diff
@@ -1,86 +1,70 @@
...
+/datum/armament_entry/company_import/deforest/medpens/emergency_pen
+	item_type = /obj/item/reagent_containers/hypospray/medipen/ekit
+
+/datum/armament_entry/company_import/deforest/medpens/blood_loss
+	item_type = /obj/item/reagent_containers/hypospray/medipen/blood_loss
+
+/datum/armament_entry/company_import/deforest/medpens/atropine
+	item_type = /obj/item/reagent_containers/hypospray/medipen/atropine
+
+/datum/armament_entry/company_import/deforest/medpens/roxxane
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/company_imports/code/armament_datums/deforest_medical.dm]:
```suggestion
/datum/armament_entry/company_import/deforest/medpens/oxandrolone
```

File: modular_skyrat/modules/company_imports/code/armament_datums/bolt_nanotrasen_firearms.dm [hunk 5]
```diff
@@ -10,92 +10,103 @@
 /datum/armament_entry/company_import/nanotrasen_bolt_weapons/armor/slim_vest
 	name = "type I vest - slim"
 	item_type = /obj/item/clothing/suit/armor/vest
-	lower_cost = CARGO_CRATE_VALUE
-	upper_cost = CARGO_CRATE_VALUE * 1.5
-	stock_mult = 2
+	cost = PAYCHECK_CREW * 3
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/company_imports/code/armament_datums/bolt_nanotrasen_firearms.dm:13]: Is it possible for this to cause issues due to the value of the crates they come in?

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/company_imports/code/datums/cargo_pack.dm (hunk 1 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/company_imports/code/datums/cargo_pack.dm:5]: Items only come as crates once more than five of them are ordered, and a fee for the crate is automatically charged to the buyer's account.

File: tgui/packages/tgui/interfaces/CargoImportConsole.js (hunk 2 above)
Comment (Paxilmaniac) [reply tgui/packages/tgui/interfaces/CargoImportConsole.js:183]: Ammo is now listed alongside the weapons rather than in the menu. This allows for future plans especially but mainly lets me easier sell multiple different types of magazines for the same weapon without having to duplicate all of the code for buying stuff. If that doesn't make sense trust me the magazine buying thing made less sense.

File: modular_skyrat/modules/company_imports/code/armament_datums/kahraman_industries.dm (hunk 3 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/company_imports/code/armament_datums/kahraman_industries.dm:8]: OSHA is an american only organisation where as FOHSA (made up on the spot) for Frontier Occupational Health & Safety Administration is a bit more applicable to the setting.

File: modular_skyrat/modules/company_imports/code/armament_datums/deforest_medical.dm (hunk 4 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/company_imports/code/armament_datums/deforest_medical.dm]: No fun allowed.

File: modular_skyrat/modules/company_imports/code/armament_datums/bolt_nanotrasen_firearms.dm (hunk 5 above)
Comment (Paxilmaniac) [reply modular_skyrat/modules/company_imports/code/armament_datums/bolt_nanotrasen_firearms.dm:13]: Basically what I said earlier, if they're ordered with the cargo budget they come in valueless large wooden crates that need to be crowbarred. If they are private ordered, they come in a locked briefcase until there are more than five items in which case it'll come in a crate and charge you for the crate as well.

**Comment by HoodMilk** [comment]
From the round it was testmerged on, stuff was way too cheap. People were getting peacekeeper shotguns for 400 credits and it was madness. All of the other parts of this PR, like items being merged and stock being gone, are good though. Just needs some price adjustments and it's golden.

**Comment by chemistrymain2** [comment]
> From the round it was testmerged on, stuff was way too cheap. People were getting peacekeeper shotguns for 400 credits and it was madness. All of the other parts of this PR, like items being merged and stock being gone, are good though. Just needs some price adjustments and it's golden.

Cargo could already do this.

The guns were never expensive. They always cost much much less than even regular laser crates.

**Comment by ghost** [comment]
Great change. Ditching limited stocks means cargo and command players can get a bit more involved during emergencies and start mass-ordering/distributing weapons to the crew when station-wide threats like blob or pirates spawn. This is something that used to be relatively commonplace before gun-cargo; but ironically the removal of TG's gun crates in favor of limited availability firearms in company imports actually made it significantly harder to arm the crew in a pinch.

Interest has always just been kind of an unfun and unnecessary system that forces players to blow credits on random bullshit that they'll immediately throw out for no real reason - I don't think many will miss it.

**Comment by ghost** [comment]
image
**Any** crew being able to purchase an AMR from the start of the shift (if funds permit, or they pool cash) is entirely unacceptable. All weapons need to be access-locked, as a few of them already are.

**Comment by burgerenergy** [comment]
I kind of feel like the prices are _too_ cheap. Like, a MOD suit core probably shouldn't cost the same as a box of raisins elsewhere on the station. There's a fine line between available and insignificant, and I think we have veered into the the latter. As a suggestion, unfortunately a kind of time consuming one, double it across the board and then check anything over a thousand to see if it needs to be that expensive.

While you're here tho could you slim the WT-550 Autorifle's description down so it fits in the standard window, god bless.

**Comment by Paxilmaniac** [comment]
> image
> **Any** crew being able to purchase an AMR from the start of the shift (if funds permit, or they pool cash) is entirely unacceptable. All weapons need to be access-locked, as a few of them already are.

forgot to tag the rifles in that tab as restricted i can fix that. if someone is gathering 1200 credits at the /start of the round/ like you say then brother they deserve to buy whatever the hell they want

**Comment by Krypandenej** [comment]
Wouldn't making the imports even cheaper/easier to earn stack the game even FURTHER against the few antags/OPFORs we get? Nobody's going to want to even try being an actual violent villain with a typed backstory when the crew can participate in advanced validhunting and our 8(!!) sec members can immediately get their funny MCRs.

**Comment by chemistrymain2** [comment]
> image **Any** crew being able to purchase an AMR from the start of the shift (if funds permit, or they pool cash) is entirely unacceptable. All weapons need to be access-locked, as a few of them already are.

If they weren't access locked before, I don't see why they'd be access locked now.

**Comment by Namwty** [comment]
I do hate buying 5 "NRI" suits just to be able to access a out of stock GZ-03 armor vest, which then get's new stock, but company interest falls off, which means I have to repeat the cycle.

I am bit worried about some item pricing maybe being too low?
Biggest one being Bruise pack, that thing being cheap will kill need for any other medical stuff really, when it comes to brute.
1 Bruise kit action = 40 brute healed ( Then it goes on "cooldown of sorts" where it's healing is ass for some time, and it slowly goes back to 40 healing )

Now doesn't sound too bad, but then you realize you can actually grind bruise packs ( same with ointments for lenturi, but at least ointments don't heal burns like bruise pack does ) into 75% purity libital. 1 Bruise pack = 10 units libital.

So now you will have people buying super cheap ( cheaper than medikits it seems ) bruise packs, and getting 40 brute heal action + 50 libital units a pack.


But again, we don't really have to worry too much about balancing ( we aren't TG so ), so it's not too much of a big deal, just something I wanted to point out.

**Comment by ApplePlastic** [comment]
Hi, cargo main here. The only thing I don't like about this PR is the removal of needing to buy the companies before accessing their catalog. 

Please Fix. Thank You.

**Comment by dolphinemote** [comment]
Good PR.
Energy rifles seem to be really cheap though.

**Comment by KathrinBailey** [comment]
No item stock also means this is a source of near-free insuls for the entire crew. They're so cheap that most roles can buy them with their round-start cash.

**Comment by chemistrymain2** [comment]
> No item stock also means this is a source of near-free insuls for the entire crew. They're so cheap that most roles can buy them with their round-start cash.

This is, again, not really a problem, because you could always order them from cargo anyway and they'd buy it if you asked.

**Comment by burgerenergy** [comment]
The point is that insulated gloves came from imports, which had a stock, so only three~ people could get gloves.

**Comment by Paxilmaniac** [comment]
> No item stock also means this is a source of near-free insuls for the entire crew. They're so cheap that most roles can buy them with their round-start cash.

Insuls are such a skyrat non-issue that I could make them literally free and you wouldn't see a difference. This isn't tgstation sybil where everyone in a grey jumpsuit is stripping any place they can get into bare of anything valuable. People outright don't tide here and there are very very few situations where the outcome would be changed because you have yellows. You get the chunky fingers trait with them on remind you, no guns no using your pda, so on, I would have changed it already if people WERE buying free insuls instantly every round no matter what, but the truth is that it just doesn't happen. In fact, I don't think there have been very many orders for insulated gloves at all actually. If the number isn't zero, then its single digits.

**Comment by chemistrymain2** [comment]
> The point is that insulated gloves came from imports, which had a stock, so only three~ people could get gloves.

I understand that point.

I _don't_ understand, though, what the problem is. We don't have antags. We _won't_ have antags, and most especially we won't have malf AIs shocking doors, because that gets you in trouble-and we _**most especially**_ won't have malf AIs shocking doors all over the station to pen in the entire crew wantonly, which is typically going to be the #1 thing 'everyone has insulated gloves' would cockblock.

The theoretical balance problems this creates don't exist because we don't have the antags that'd cause the balance problem in the first place. Any crewmember who wanted gloves could _get_ insulated gloves already in a two hour thirty minute round with minimal effort-it was as much as saying 'cargo, order some insulated gloves for me please' or buying a private crate, which you could _already do_, and getting insulated gloves that way.

It wasn't an expensive purchase. It wasn't even a hard to get purchase. You could go to cargo, click 'buy privately', and buy insulated gloves at any point during our two hour, thirty minute round.

That the entire crew can get insulated gloves if they feel like it is no different to how it already was beforehand, just slightly easier and cheaper, and it being cheaper/easier doesn't matter because _we don't have antags that can shock doors_ and we don't have situations where having insulated gloves is powerful/obligatory.

And because of this + the fact that insulated gloves give you chunky fingers, nobody's going to buy those roundstart to start with because everyone _already _ didn't buy those roundstart to start with.


Yes, there's the theoretical scenario where a malf AI gets doomsday or something, bolts everything down, and Oh no! Everyone has insulated gloves, I guess i'm powerless-

But this would never happen, because we don't have antags, we don't have malf, we don't have doomsday, we don't have rules that permit the AI to shockbolt doors everywhere, and nobody would buy the gloves anyway. 

This is what I mean when I keep saying 'we're not TG and don't have to balance like TG'-we face none of the same problems or the 'competitive server' elements. There's maybe one (1) bad guy a round and it's probably pirates or a contractor. Nobody is competing against anyone. This doesn't have balance issues because environmental hazard balance concerns are a nonissue.

(and yes, chunky fingers is enough of a downside people don't seek out insuls unless they have to. If these were combat gloves i'd raise an eyebrow.)

**Comment by burgerenergy** [comment]
I don't particularly think Insuls are an issue, no. I do roll my eyes whenever I seem to assistant wearing them tho

fucking assistants

It is however an item to make basically anyone a full engineer in an instant, so if we get lots of cases of Joe Shmoe the super doctor or something it might be worth looking into in lieu of a never-gonna-happen skill system, but eh. That's a big hypothetical.

**Comment by burgerenergy** [comment]
alright having experienced it in a round;

There needs to be a regenerating limit to weapons. We had a red alert and a self-funded cargo bought twenty laser rifles to just give them away. That sort of thing shouldn't be possible, as even ignoring any semblance of crew vs antag balance is tips the action scale to turbo high lethality, which not even TG is about.

**Comment by Stalkeros2** [comment]
The exact thing can be done even without the company imports just by ordering armory crates that are also limitless.

**Review by GoldenAlpharex — APPROVED** [review]
I think it's been working fine so far, we can see about making more changes in the future as need be.
